### PR TITLE
Migrate ProductReferenceResolver

### PR DESCRIPTION
**Pull request by lojzatran**
#### Summary

Migrate ProductReferenceResolver and related tests.

### CODE REVIEW

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+        Locale.ENGLISH,
+        OLD_CATEGORY_CUSTOM_TYPE_NAME,
+        CTP_SOURCE_CLIENT);
+
+    ensureCategories(CTP_TARGET_CLIENT, getCategoryDrafts(null, 2));
+    categoryReferencesWithIds =
+        getReferencesWithIds(ensureCategories(CTP_SOURCE_CLIENT, getCategoryDrafts(null, 2)));
+
+    ensureProductType(PRODUCT_TYPE_RESOURCE_PATH, CTP_TARGET_CLIENT);
+    //    ensureProductType(PRODUCT_TYPE_NO_KEY_RESOURCE_PATH, CTP_TARGET_CLIENT);
```
Comment (salander85) [inline src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java:95]: Remove this comment

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+        CTP_SOURCE_CLIENT
+            .productProjections()
+            .get()
+            .withStaged(true)
+            .execute()
+            .thenApply(ApiHttpResponse::getBody)
+            .thenApply(ProductProjectionPagedQueryResponse::getResults)
+            .join();
+
+    final List<ProductDraft> productDrafts = // mapToProductDrafts(products);
```
Comment (salander85) [inline src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java:248]: This is commented multiple times. I think you can remove it?

**Review by salander85 — APPROVED** [review]

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java (hunk 2 above)
Comment (lojzatran) [reply src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java]: https://github.com/commercetools/commercetools-sync-java/pull/937/commits/ce838b0eb9845530ca81fce9fc944139cebf80d0

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java (hunk 1 above)
Comment (lojzatran) [reply src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/products/ProductReferenceResolverIT.java]: https://github.com/commercetools/commercetools-sync-java/pull/937/commits/ce838b0eb9845530ca81fce9fc944139cebf80d0

**Pull request merged**